core/overdrive.py

```diff
@@ -677,14 +677,14 @@ def make_link_safe(cls, url: str) -> str:
     def _do_get(self, url: str, headers, **kwargs) -> Response:
         """This method is overridden in MockOverdriveAPI."""
         url = self.endpoint(url)
-        kwargs["max_retry_count"] = self._configuration.max_retry_count
+        kwargs["max_retry_count"] = int(self._configuration.max_retry_count)
```

Comment:
I've moved this typecasting and error management to the ConfigurationGrouping class. int typecasting will still have to be done in case the float's trailing 0 is unrequired

Reply:
Yeah, I‘d also consider that a programming error and agree that it would be less likely. My main goal is being able to quickly understand what is happening if that occurred. 

For example, if we configured a bad default for a particular setting, but we didn’t happen see an error for some time because users always configured a value.

Another example that applies more directly to this case: It might make sense to configure a default of `None` or empty string to get an underlying default behavior (in this case, the default max retry count for the stack).